File: src/octave/negative.m
function negative( orig_file_name )
    %NEGATIVE creează și afișează negativul imaginii primite

    orig_img = imread(orig_file_name);
    orig_pixels = cast(orig_img,'double');
    orig_pixels = orig_pixels/255.0;

    neg_pixels = orig_pixels;

    %% neg_pixels conține valori în intevalul [0,1]

    %% Exercițiul 1: negativul imaginii originale
    %% Exercițiul 1: completați aici

    %% Exercițiul 1: ----------

    neg_pixels = round(neg_pixels*255);
    neg_img = cast(neg_pixels,'uint8');
    neg_file_name = [orig_file_name(1:(size(orig_file_name,2)-4)) '_neg.jpg'];
    image(neg_img);
    imwrite(neg_img, neg_file_name);

end

%% Tudor Berariu, 2014

File: src/cc/negative.cc
// Tudor Berariu, 2014

#include <string>

#include "display_image.h"
#include "image_data.h"
#include "jpeg_io.h"

#include <gtkmm/application.h>
#include <gtkmm/window.h>
#include "image_widget.h"


void negative(ImageData*& image) {
  // Exercițiul 1
}


int main(int argc, char** argv)
{
  std::string orig_file_name(argv[1]);
  std::string neg_file_name =
    orig_file_name.substr(0, orig_file_name.length()-4) + "_neg.jpg";

  ImageData* img;
  read_jpeg_file(orig_file_name, img);
  negative(img);
  write_jpeg_file(neg_file_name, img);

  return display_image(argv, neg_file_name, "negative");
}
